smoke_tests/run_smoke_tests.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+
+DOCKER_IMAGE=rig:smoke_test
+COMPOSE="docker-compose -f smoke_tests.docker-compose.yml"
+
+function section_header() {
+  echo ""
+  echo "╓─────[ $@ ]─────╖"
+  echo ""
+}
+
+section_header "Building docker image: ${DOCKER_IMAGE}"
+docker build -f ../smoke_tests.dockerfile -t "${DOCKER_IMAGE}" ..
+
+section_header "Starting Kafka and backend-simulation services"
+$COMPOSE up -d kafka rest-api
+
+section_header "Running tests, plaintext Kafka connection"
+$COMPOSE run --rm \
+  -e KAFKA_ENABLED=1 \
+  -e KAFKA_SSL_ENABLED=0 \
+  -e KAFKA_HOSTS=kafka:9292 \
+  rig
+
+section_header "Running tests, encrypted Kafka connection"
+$COMPOSE run --rm \
+  -e KAFKA_ENABLED=1 \
+  -e KAFKA_SSL_ENABLED=1 \
+  -e KAFKA_HOSTS=kafka:9393 \
+  -e KAFKA_SSL_KEYFILE_PASS=abcdef \
```

Comment:
Shouldn't this be `abcdefgh`? Or Am I missing something?

Reply:
Indeed, I've just discovered an error in the script that generates the certificates, which caused the private to be stored _unencrypted_. This is why the wrong password still worked, and I'm glad that you've actually catched that. Turns out that `openssl genpkey` simply ignores the `-pass` parameter if there is no algorithm selected. When I wrote this, I foolishly assumed that either an algorithm would be selected by default, or that there would be a warning in case I'd have to select one. `openssl` being `openssl`, I guess.